composition-js/src/merging/merge.ts

```diff
@@ -803,19 +824,123 @@ class Merger {
     return this.metadata(sourceIdx).isFieldFullyExternal(field);
   }
 
-  private withoutExternal(sources: (FieldDefinition<any> | undefined)[]): (FieldDefinition<any> | undefined)[] {
+  private withoutExternal(sources: FieldOrUndefinedArray): FieldOrUndefinedArray {
     return sources.map((s, i) => s !== undefined && this.isExternal(i, s) ? undefined : s);
   }
 
-  private hasExternal(sources: (FieldDefinition<any> | undefined)[]): boolean {
+  private hasExternal(sources: FieldOrUndefinedArray): boolean {
     return sources.some((s, i) => s !== undefined && this.isExternal(i, s));
   }
 
   private isShareable(sourceIdx: number, field: FieldDefinition<any>): boolean {
     return this.metadata(sourceIdx).isFieldShareable(field);
   }
 
-  private mergeField(sources: (FieldDefinition<any> | undefined)[], dest: FieldDefinition<any>) {
+  private getOverrideDirective(sourceIdx: number, field: FieldDefinition<any>): Directive<any> | undefined {
+    // Check the directive on the field, then on the enclosing type.
+    const overrideDirectiveName = this.metadata(sourceIdx).overrideDirective().name;
+    const fieldOverrideDirective = field.appliedDirectives.find(d => d.name === overrideDirectiveName);
+    const parent: FieldDefinition<any> | undefined = field.parent;
+    if (parent) { // should always be true
```

Comment:
I usually prefer a `assert(parent, "should always be true")` because it avoids bumping the indentation and, probably more importantly, it does crash loudly if the assumption changes unexpectedly someday.